lib/core/engine.js

```diff
@@ -178,17 +179,21 @@ class Engine {
     this.events.on('asset-changed', addToCargo);
   }
 
+  setupCompilerAndContractsManagerService(options) {
+    this.registerModule('compiler', {plugins: this.plugins, disableOptimizations: options.disableOptimizations});
+    this.registerModule('solidity', {ipc: this.ipc, useDashboard: this.useDashboard});
+    this.registerModule('vyper');
+    this.registerModule('contracts_manager', {compileOnceOnly: options.compileOnceOnly});
+  }
```

Comment:
Notice that we aren't requesting any services at this point. This is because we're running into race conditions otherwise when relying on this service initialization. Another option would be to pass different callbacks to `setupCompilerAndContractsManagerService()` which we can then react to from the call side.

I actually prefer that, but decided to stay consistent with other `*Service()` functions within `Engine`.